lib/Upcloud/StorageApi.php

```diff
@@ -2770,136 +903,40 @@ public function listStoragesAsync()
      *
      * List of storages
      *
-     * @throws \InvalidArgumentException
-     * @return \GuzzleHttp\Promise\PromiseInterface
-     */
-    public function listStoragesAsyncWithHttpInfo()
-    {
-        $returnType = '\Upcloud\ApiClient\Model\SuccessStoragesResponse';
-        $request = $this->listStoragesRequest();
-
-        return $this->client->sendAsync($request)->then(function ($response) use ($returnType) {
-            $responseBody = $response->getBody();
-            if ($returnType === '\SplFileObject') {
-                $content = $responseBody; //stream goes to serializer
-            } else {
-                $content = $responseBody->getContents();
-                if ($returnType !== 'string') {
-                    $content = json_decode($content);
-                }
-            }
-
-            return [
-                ObjectSerializer::deserialize($content, $returnType, []),
-                $response->getStatusCode(),
-                $response->getHeaders()
-            ];
-        }, function ($exception) {
-            $response = $exception->getResponse();
-            $statusCode = $response->getStatusCode();
-            throw new ApiException(
-                "[$statusCode] Error connecting to the API ({$exception->getRequest()->getUri()})",
-                $statusCode,
-                $response->getHeaders(),
-                $response->getBody()
-            );
-        });
-    }
-
-    /**
-     * Create request for operation 'listStorages'
-     *
-     * @throws \InvalidArgumentException
-     * @return \GuzzleHttp\Psr7\Request
+     * @param string|null $type
+     * @throws InvalidArgumentException
+     * @return PromiseInterface
      */
-    protected function listStoragesRequest()
+    public function listStoragesAsyncWithHttpInfo(?string $type = null): PromiseInterface
     {
-
-        $resourcePath = '/storage';
-        $formParams = [];
-        $queryParams = [];
-        $headerParams = [];
-        $httpBody = '';
-        $multipart = false;
-
-
-
-
-        if ($multipart) {
-            $headers= $this->headerSelector->selectHeadersForMultipart(
-                ['application/json']
-            );
+        if ($type) {
+            Assert::oneOf($type, StorageType::getAllowableEnumValues());
```

Comment:
`Exception when calling StorageApi->listStorages: Expected one of: "disk", "cdrom", "template", "backup", "normal". Got: "favorite"`
Doesn't allow for all accepted request filters in [list storages](https://developers.upcloud.com/1.3/9-storages/#list-storages)

>Returns a list of all accessible storages. The list can be narrowed down by the storage's access type (public or private), storage type (normal, backup, cdrom or template) or favorite status.